src/util/mpir_hw_topo.c

```diff
@@ -696,6 +741,66 @@ MPIR_Node_obj_osdev_type MPIR_Node_get_osdev_obj_type(MPIR_Node_obj dev_obj)
     return ret;
 }
 
+int MPIR_Node_set_membind(void *baseaddr, size_t len, int num_ids, int *obj_ids,
```

Comment:
It is not merely set (as in setting an attribute), it is the actual action of binding the memory to a set of nodes. Maybe `MPIR_Node_mem_bind` is a better name?

For the common case that we want to do simple binding, we only need a single `obj_id`, right? I find putting the two policy case together a bit taxing in reading. How about split them into two functions, leave `MPIR_Node_mem_bind` to do the default `__BIND` policy, and use additional function, `MPIR_Node_mem_interleave` to do the `__INTERLEAVE` policy?

EDIT: actually `_interleave` is an internal detail. From usage point of view, it is `membind_single` and `membind_multiple` right?

Reply:
Another thing, is `MPIR_Node_` a good namespace? I don't find `Node` intuitive for its meaning. How about `MPIR_topo_`?

I know you meant this is for on-node vs `network`, but it doesn't make sense to do binding for network anyway, so having `_Node`` there is redudant.

Reply:
I am reluctant to change `MPIR_Node` to `MPIR_topo` as this would break the namespace convention.